luarules/gadgets/unit_water_depth_damage.lua

```diff
@@ -0,0 +1,179 @@
+function gadget:GetInfo()
+	return {
+		name = "Water Crush and Collision Damage",
+		desc = "Creates and handles water collision events, and kills units stuck underwater",
+		author = "SethDGamre",
+		date = "2024.9.22",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+--use customParams.water_fall_damage_multiplier = 1.0 to change the amount of fall damage taken by specific units.
+
+--any maxWaterDepth movedef equal to or above this number will not take drowning damage.
+local isDrownableMaxWaterDepth = 5000
+
+--a percentage of health taken as damage per second when stuck below maxWaterDepth.
+local drowningDamage = 0.05
+
+--base damage percentage multiplied by velocity upon impact with water.
+local fallDamage = 0.18
+
+--this influences the compounding escalation of fall damage from water collisions.
+local fallDamageCompoundingFactor = 1.05
+
+--statics
+local gameFrame = 0
+local gameFrameExpirationThreshold = 3
+local gaiaTeamID = Spring.GetGaiaTeamID()
+local velocityThreshold = 2.5
+local waterDamageDefID = -5
```

Comment:
why -5

Reply:
Oh, that's right there is a water damage defined somewhere. I will see if I can find it.

-5 was the number I got from the spring wiki for water damage

Reply:
```suggestion
local waterDamageDefID = Game.envDamageTypes.Water
```